modules/ROOT/edits/execution-engine.adoc

```diff
@@ -0,0 +1,152 @@
+
+See specific component or module documentation to learn the processing type it supports. If none is specified, the default is CPU Light.
+
+For connectors created with the Mule SDK, the SDK determines the most
+appropriate processing type based on how the connector is implemented. For
+details on that mechanism, refer to the xref:1.1@mule-sdk::index.adoc[Mule SDK documentation].
+
+[[threading]]
+== Threading
+
+Based on the processing type of a component, Mule executes that component on a thread pool that is specifically tuned for that kind of work. These thread pools are managed by Mule and shared across all apps in the same Mule instance.
+When started, Mule introspects the available resources (such as memory and CPU cores) in the system to tune thread pools automatically for the environment where it is running. The default configuration was established through performance testing, which found optimal values for most scenarios.
+
+The different thread pools allow Mule to manage resources more efficiently, requiring significantly fewer threads (and their inherent memory footprint) to handle a given workload.
+
+Following are described the key aspects of each thread pool.
+
+=== CPU Light
+
+CPU Light is for a relatively small thread pool (2 threads per available core by default).
+
+Apart from executing the CPU Light processors, this pool performs the handoff
+of the event between processors in the flow (including the routers) and the
+response handling for non-blocking IO.
+
+Sometimes, throughput might drop or become unresponsive in an app due to some code
+misusing the CPU Light thread pool. This issue can be quickly checked by taking a
+thread dump of the runtime and looking for `WAITING` or `BLOCKED` or for
+long-running processes in the CPU Light threads.
+
+=== CPU Intensive
+
+CPU Intensive is also a small thread pool (2 threads per available core by default),
+but it provides a queue for accepting more tasks.
+
+=== IO
+
+IO is an elastic thread pool that grows as needed.
+
+Tasks running in this pool should spend most of their time `WAITING` or `BLOCKED`
+instead of doing CPU work, so they do not compete with the other pools.
+
+Also, when a transaction is active (since many transaction managers require all
+steps of the same transaction to be handled by the same thread), the `IO` pool
+is used.
+
+=== Custom Pools
+
+Apart from the three core thread pools, Mule or some components might create additional pools for specific purposes, such as:
+
+* NIO Selectors +
+Enables non-blocking IO. Each connector can create as many as required.
+* Recurring tasks pools +
+Some connectors or components might create specific pools to perform recurring tasks (e.g: expiration monitors, queue consumers, etc.)
+
+=== Proactor Pattern
+
+Proactor is a design pattern for asynchronous execution. To understand how the Proactor design pattern works, visit https://en.wikipedia.org/wiki/Proactor_pattern
```

Comment:
```suggestion
Proactor is a design pattern for asynchronous execution. To understand how the Proactor design pattern works, visit https://en.wikipedia.org/wiki/Proactor_pattern.
```

Reply:
I wonder if we really need this cross-ref right at the beginning of this topic, since we subsequently define the relevant info about how this works in this context?
Consider moving to the end of the topic, introducing with something like "For more information...".